### PR TITLE
Fix slot binding in chest or non-player GUIs

### PR DESCRIPTION
**Pull request by IndigoPolecat**
Fix a desync between the event slot number and the gui slot number.

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/SlotLocking.java [hunk 1]
```diff
@@ -473,8 +473,20 @@ public void onWindowClick(SlotClickEvent slotClickEvent) {
...
+			if (!(Minecraft.getMinecraft().currentScreen instanceof GuiContainer)) {
+				return;
+			}
+			GuiContainer container = (GuiContainer) Minecraft.getMinecraft().currentScreen;
+			int size = container.inventorySlots.inventorySlots.size();
+
 			int from, to;
 			int id = slotClickEvent.slot.getSlotIndex();
+			// if the inventory is bigger than the vanilla inventory
+			// I don't know of a way to detect non-player inventories but it should to apply here and L505
```
[the remark this answers is not in the document]
Comment (IndigoPolecat) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/SlotLocking.java]: it was a mistake from another bug I tried to fix but was unable to, I removed the line I was referring to already and I added a commit that adjusted the comment a minute ago

**Comment by NopoTheGamer** [comment]
Does this fix https://github.com/NotEnoughUpdates/NotEnoughUpdates/issues/519 ?

**Comment by IndigoPolecat** [comment]
> Does this fix #519 ?

I didn't realise there was an issue open but yes it does.
I also just added a commit to remove a very dumb assumption I had made which means it will now work in any situation, regardless of if the total inventory has less than 45 slots (like a furnace).

**Review by NopoTheGamer — CHANGES_REQUESTED** [review]
If you shift click from hotbar it will just go into the chest and then the link will be broken once you exit the chest. Slot binding just full doesn't work either side in the furnace, did notice it works perfectly in the dropper

**Comment by IndigoPolecat** [comment]
> If you shift click from hotbar it will just go into the chest and then the link will be broken once you exit the chest. Slot binding just full doesn't work either side in the furnace, did notice it works perfectly in the dropper

That's odd, I tested in a furnace but I'll double check, I think the other issue was pre existing but I'll look into it as well

**Comment by IndigoPolecat** [comment]
> > If you shift click from hotbar it will just go into the chest and then the link will be broken once you exit the chest. Slot binding just full doesn't work either side in the furnace, did notice it works perfectly in the dropper
> 
> That's odd, I tested in a furnace but I'll double check, I think the other issue was pre existing but I'll look into it as well

I can confirm that shift clicking the hotbar was an issue prior to my changes, I will make another attempt at fixing it but I cannot be certain I will be able to.

As far as I can tell the only issue with furnace (and it also applies to other guis like chests) is spam clicking which I _think_ causes the shift click to be processed by minecraft once the As far as I can tell the only issue with furnace (and it also applies to other guis like chests) is spam clicking which I _think_ causes the shift click to be processed by minecraft once the item is moved to the hotbar. I have no idea how to affect this with how the current code works though since the event is different from minecraft's and can't be easily cancelled (plus I'd be worried about potentially cancelling packets w/ watchdog).item is moved to the hotbar. I have no idea how to affect this with how the current code works though since the event is different from minecraft's and can't be easily cancelled (plus I'd be worried about potentially cancelling packets w/ watchdog).

**Review by NopoTheGamer — APPROVED** [review]

**Review by nea89o — APPROVED** [review]

**Pull request merged**